### ISSUE TITLE
Content Offset wrong after dismissing modal ViewController

### ISSUE THREAD

**Issue by Siemian** [MessageKit/MessageKit#1001]
## Problem
After presenting modal View Controller and going back to Messages View Controller Scroll View offset is wrong.
### Conditions
This issue occurs when we use a little more complicated layout. In my case we had MessagesViewController as a child view controller, where constraints were created from code and `translatesAutoresizingMaskIntoConstraints` is set to false on `MessagesViewController.view`.

Video: https://take.ms/Ul45v

### Reproduction steps
1. Download source code from this fork: https://github.com/Siemian/MessageKit/tree/layoutIssue
[Diff Link](https://github.com/MessageKit/MessageKit/compare/master...Siemian:layoutIssue)
2. Start app
3. Click on any message
4. Close modal by clicking red square
5. Offset is wrong

#### 1
If we use source code provided in the fork and set [translatesAutoresizingMaskIntoConstraints](https://github.com/MessageKit/MessageKit/compare/master...Siemian:layoutIssue#diff-930b6ee73687964fabb525c330a64213R44) to true, it will work as expected. But when you have app where constraints are created from code setting `translatesAutoresizingMaskIntoConstraints` is not possible solution.
#### 2
When app is started offset is right, at this time `handleKeyboardDidChangeState` is called for the first time and we can observe values: 
![zrzut ekranu 2019-02-26 o 17 06 06](https://user-images.githubusercontent.com/1449108/53427294-fb21c100-39e8-11e9-9e4b-0d0ae656cb5a.png)

After dismissing modal view `handleKeyboardDidChangeState` is called again. Now values are different. 
![zrzut ekranu 2019-02-26 o 17 07 44](https://user-images.githubusercontent.com/1449108/53427374-20aeca80-39e9-11e9-9429-fd11f2950a5c.png)
Those new values are wrong.

- What version of MessageKit are you using?
2.0.0
- What version of iOS are you running on?
11 & 12
- What version of Swift are you running on?
4.2
- What device(s) are you testing on? Are these simulators?
Simulator & Device
- Is the issue you're experiencing reproducable in the example app?
Only after some modifications from the fork.

**Comment by nathantannar4** [comment]
Possible duplicate of #993 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Siemian** [comment]
I've checked this issue and by looking at the description:
> Following these steps to reproduce, you will see that only the discarded notification is sent, thus leaving the bottom content inset unchanged causing an overlap:

I think this case is different because here, delivered notification is processed and not discarded.
